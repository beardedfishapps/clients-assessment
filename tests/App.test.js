import { describe, test, expect } from 'vitest'
import App from '../app.vue'

/**
 * I normally would write several test functions to test different parts of the component,
 * passing different parameters, verifying the component mounts, etc.
 *
 * However it appears that the testing utils with Nuxt3 are not ready to be used, and I spent
 * too much time trying to get Jest to work. Previously I have written several dozen test cases
 * for Vue2/Nuxt2, so I am at a bit of a loss as to how this is the way it is.
 */
describe('Clients Assessment Tests', async () => {
  test('test that App renders', () => {
    const result = App
    expect(result).toMatchSnapshot()
  })
})
