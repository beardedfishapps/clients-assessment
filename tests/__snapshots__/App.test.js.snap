// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Clients Assessment Tests > test that App renders 1`] = `
{
  "__file": "/home/banjokaboom/Documents/dev/context-labs-assessment/clients-assessment/app.vue",
  "__name": "app",
  "setup": [Function],
  "ssrRender": [Function],
}
`;
